src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs

```diff
@@ -15,46 +19,92 @@ internal sealed class DeclarationTable
     /// Constructs a new declaration table from the given syntax trees.
     /// </summary>
     /// <param name="syntaxTrees">The syntax trees to construct the declarations from.</param>
+    /// <param name="rootPath">The path to the root module.</param>
     /// <returns>The declaration table containing <paramref name="syntaxTrees"/>.</returns>
-    public static DeclarationTable From(ImmutableArray<SyntaxTree> syntaxTrees) => new(syntaxTrees);
+    public static DeclarationTable From(ImmutableArray<SyntaxTree> syntaxTrees, string rootPath, Compilation compilation) => new(syntaxTrees, rootPath, compilation);
 
     /// <summary>
     /// An empty declaration table.
     /// </summary>
-    public static DeclarationTable Empty { get; } = new(ImmutableArray<SyntaxTree>.Empty);
+    public static DeclarationTable Empty { get; } = new(ImmutableArray<SyntaxTree>.Empty, string.Empty, Compilation.Create(ImmutableArray<SyntaxTree>.Empty));
 
     /// <summary>
     /// The merged root module.
     /// </summary>
     public MergedModuleDeclaration MergedRoot => this.mergedRoot ??= this.BuildMergedRoot();
     private MergedModuleDeclaration? mergedRoot;
 
+    public string RootPath { get; }
+
+    private readonly Compilation compilation;
```

Comment:
Do we only take a compilation for the diagnostic bag? If so, let's just take the diagnostic bag directly

Reply:
On the other hand, we also take the root path that Compilation should already have, if we take a compilation, we shouldn't take a root path, so maybe just remove taking the root path and do something like `public string RootPath => this.compilation.RootPath`